### PR TITLE
Fixes #8923: Requires Java8 (jdk8) for Rudder 4.0

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8923

### CODE REVIEW

File: packages.makefile [hunk 1]
```diff
@@ -35,10 +35,10 @@ BUILDREQUIRESFEDORA := $(shell grep -s "FEDORA${OSVERSION}" SOURCES/.dependencie
 JAVAREQUIRES := $(shell grep -s "JAVA" SOURCES/.dependencies | cut -d ':' -f2)
 
 # Original URL: http://www.oracle.com/technetwork/java/javase/downloads/index-jsp-138363.html#javasejdk
-JDKURL := http://www.normation.com/tarball/java/jdk-7u71-linux-i586.rpm
+JDKURL := http://www.normation.com/tarball/java/jdk-latest-linux-i586.rpm
```
Comment (jooooooon) [inline packages.makefile]: This looks like a symbolic URL that will provide a different version at different points of time. We must not use such "moving" versions, because it means we don't know how to rebuild a given release, which may make reproducing (and therefore fixing) bugs impossible.

File: rudder-webapp/debian/control [hunk 2]
```diff
@@ -2,13 +2,13 @@ Source: rudder-webapp
...
 Maintainer: Rudder packaging team <rudder-packaging@rudder-project.org>
-Build-Depends: debhelper (>= 7), openjdk-7-jdk | openjdk-8-jdk
+Build-Depends: debhelper (>= 7), openjdk-8-jdk
 Standards-Version: 3.8.0
 Homepage: http://www.rudder-project.org
 
 Package: rudder-webapp
 Architecture: all
-Depends: ${shlibs:Depends}, ${misc:Depends}, rudder-jetty | jetty | jetty8, rudder-techniques (>= ${binary:Version}), apache2, apache2-utils, ncf, git-core, rsync, lsb-release, openssl, ldap-utils, postgresql-client (>=8.4), ncf-api-virtualenv
+Depends: ${shlibs:Depends}, ${misc:Depends}, rudder-jetty (= ${binary:Version}) | jetty | jetty8, rudder-techniques (>= ${binary:Version}), apache2, apache2-utils, ncf, git-core, rsync, lsb-release, openssl, ldap-utils, postgresql-client (>=8.4), ncf-api-virtualenv, java8-runtime | openjdk-8-jre
```
Comment (jooooooon) [inline rudder-webapp/debian/control]: This looks very much like an unrelated change `rudder-jetty (= ${binary:Version}) ` - if it's missing, please fix it in a dedicated ticket, on all branches.

**Comment by peckpeck** [comment]
Commit modified
